product_contract_recurrence_in_price/models/sale_order_line_contract_mixin.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2025 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLineContractMixin(models.AbstractModel):
+    _inherit = "sale.order.line.contract.mixin"
+
+    include_recurrence_in_price = fields.Boolean(
+        help="The amounts of the Sale Order Line will be multiplied"
+        "by the number of times the line will be invoiced.",
+        compute="_compute_product_contract_data",
+        precompute=True,
+        store=True,
+        readonly=False,
+    )
+    recurring_invoicing_number = fields.Float(
+        compute="_compute_recurring_invoice_number",
+        digits=(16, 1),
+        help="Number of times a line will be invoiced.",
+        store=True,
+        readonly=False,
+        precompute=True,
+    )
+
+    @api.depends("product_id")
+    def _compute_product_contract_data(self):
+        res = super()._compute_product_contract_data()
+        for rec in self:
+            if rec.product_id.is_contract:
+                rec.include_recurrence_in_price = (
+                    rec.product_id.include_recurrence_in_price
+                )
```

Comment:
You aren't setting your field in every cases
```suggestion
                rec.include_recurrence_in_price = rec.product_id.is_contract and (
                    rec.product_id.include_recurrence_in_price
                )
```

Reply:
Done :+1: